addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-gcp/src/test/java/org/nuxeo/ecm/core/storage/gcp/TestGoogleBinaryManager.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+package org.nuxeo.ecm.core.storage.gcp;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assume.assumeFalse;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.blob.AbstractCloudBinaryManager;
+import org.nuxeo.ecm.blob.AbstractTestCloudBinaryManager;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+import com.google.cloud.storage.StorageException;
+
+/**
+ * WARNING: You must pass those variables to your test configuration:
+ * <p/>
+ * <pre>
+ *   -Dnuxeo.gcp.project: GCP Project ID
+ *   -Dnuxeo.gcp.storage.bucket: GCP Bucket ID
+ *   -Dnuxeo.gcp.credentials: GCP Credentials
+ * </pre>
+ * <p/>
+ * Don't set any bucket prefix (nuxeo.gcp.storage.bucket_prefix) as the unit tests don't check it.
+ */
+@RunWith(FeaturesRunner.class)
+@Features(RuntimeFeature.class)
+public class TestGoogleBinaryManager extends AbstractTestCloudBinaryManager<GoogleStorageBinaryManager> {
+
+    protected final static List<String> PARAMETERS = Arrays.asList(GoogleStorageBinaryManager.BUCKET_NAME_PROPERTY,
+            GoogleStorageBinaryManager.PROJECT_ID_PROPERTY, GoogleStorageBinaryManager.GOOGLE_APPLICATION_CREDENTIALS);
+
+    protected static Map<String, String> properties = new HashMap<>();
+
+    @BeforeClass
+    public static void initialize() {
+        AbstractCloudBinaryManager bm = new GoogleStorageBinaryManager();
+        PARAMETERS.forEach(s -> properties.put(s, Framework.getProperty(bm.getSystemPropertyName(s))));
+        // Ensure mandatory parameters are set
+        PARAMETERS.forEach(s -> assumeFalse(isBlank(properties.get(s))));
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        // Cleanup keys
+        Properties props = Framework.getProperties();
+        PARAMETERS.forEach(props::remove);
+    }
+
+    @Test
+    public void iCanCreateAndDeleteBuckets() {
+        assertThat(binaryManager.getOrCreateBucket(binaryManager.bucketName)).isNotNull();
+        assertThat(binaryManager.deleteBucket(binaryManager.bucketName)).isTrue();
+    }
+
+    @Override
+    protected GoogleStorageBinaryManager getBinaryManager() throws IOException {
+        GoogleStorageBinaryManager binaryManager = new GoogleStorageBinaryManager();
+        binaryManager.initialize("gcptest", properties);
+        return binaryManager;
+    }
+
+    @Override
+    protected Set<String> listObjects() {
```

Comment:
Please make this method take into account the prefix too. See `AbstractS3BinaryTest.listObjects`. You may also want a separate `listAllObjects`, see in the same class.